### PR TITLE
Implement ja augment synonym

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation

Implement Japanese version of `augment.synonym()`

### CODE REVIEW

**Comment by Alnusjaponica** [comment]
@liwii 
Fixed all CI errors. I'm going to update the docstrings and will make this PR review ready.

File: pyproject.toml [hunk 1]
```diff
@@ -26,6 +27,8 @@ dependencies = [
     'rouge_score >= 0.1.2',
     'sentence_transformers >= 2',
     'sentencepiece>=0.1.95',
+    'sudachidict_core',
+    'sudachipy',
```
Comment (liwii) [inline pyproject.toml:31]: Building `sudachipy` requires a Rust compiler. Could you add that to the README or move the requirement into optional?

Comment (liwii) [reply pyproject.toml:31]: Or we can create a new section like `ja`

File: src/langcheck/augment/ja/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from langcheck.augment.ja._synonym import synonym
+
+__all__ = ["synonym"]
```
Comment (liwii) [inline src/langcheck/augment/ja/__init__.py:3]: Could you add "ja" to `langcheck.augment` too?

File: src/langcheck/augment/ja/_synonym.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    perturbed_instances = []
+    for instance in instances:
+        tokens = sudachi_tokenizer.tokenize(instance)
+        for _ in range(num_perturbations):
+            perturbed_instance = ""
+            for token in tokens:
+                synonym = token.surface()
+                if (synonyms := chikkar.find(token.normalized_form())
+                   ) and random.random() < kwargs["aug_p"]:
+                    while synonym == token.surface():
```
Comment (liwii) [inline src/langcheck/augment/ja/_synonym.py:47]: Isn't it better to just remove `token.surface()` from `synonyms` and call `random.choice` only once?

File: tests/augment/ja/test_synonym.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        (["地球は青かった。", "先行きが不安だ。"
+         ], 2, ["アースは青かった。", "アースは青かった。", "フューチャーが気掛かりだ。", "行く先が畏れだ。"]),
+    ],
+)
+def test_synonym(instances: list[str] | str, num_perturbations: int,
+                 expected: list[str]):
+    seed = 42
+    random.seed(seed)
+    actual = synonym(instances, num_perturbations=num_perturbations)
+    print(actual)
```
Comment (liwii) [inline tests/augment/ja/test_synonym.py:27]:
```suggestion
```

**Review by liwii — COMMENTED** [review]
Feature-wise LGTM, made a few minor comments.

File: src/langcheck/augment/ja/_synonym.py (hunk 3 above)
Comment (Alnusjaponica) [reply src/langcheck/augment/ja/_synonym.py]: It seems `chikkar.find()` result doesn't include original word, so I simply removed while loop.

File: pyproject.toml (hunk 1 above)
Comment (Alnusjaponica) [reply pyproject.toml]: I put them in the [optional] dependency.

**Comment by Alnusjaponica** [comment]
@liwii Fixed CI failures. PTAL.

**Review by liwii — APPROVED** [review]

**Pull request merged**